libhwmon/wrapadl.cpp

```diff
@@ -197,7 +263,8 @@ int wrap_adl_get_fanpcnt(wrap_adl_handle *adlh, int gpuindex, unsigned int *fanp
 
 	ADLFanSpeedValue *fan = new ADLFanSpeedValue();
 	fan->iSpeedType = 1;
-	rc = adlh->adlOverdrive5FanSpeedGet(adlh->phys_logi_device_id[gpuindex], 0, fan);
+	int adlidx = adlh->opencl_adl_device_id[gpuindex];
```

Comment:
gpuindex should be always adl id, the caller should use the mapping array if needed